yb-voyager/src/cp/yugabyted/yugabyted.go

```diff
@@ -0,0 +1,621 @@
+	}
+}
+
+// Send visualisation metadata
+func (cp *YugabyteD) sendVisualizerDBPayload(
+	visualizerDBPayload VisualizerDBPayload) error {
+	cmd := fmt.Sprintf("INSERT INTO %s ("+
+		"migration_uuid, "+
+		"migration_phase, "+
+		"invocation_sequence, "+
+		"migration_dir, "+
+		"database_name, "+
+		"schema_name, "+
+		"payload, "+
+		"db_type, "+
+		"status, "+
+		"invocation_timestamp"+
+		") VALUES ("+
+		"$1, $2, $3, $4, $5, $6, $7, $8, $9, $10"+
+		")", YUGABYTED_METADATA_TABLE_NAME)
+
+	var maxAttempts = 5
+	visualizerDBPayload.MigrationDirectory = cp.migrationDirectory
+
+	var err error
+	for attempt := 1; attempt <= maxAttempts; attempt++ {
+		err = cp.executeInsertQuery(cmd, visualizerDBPayload)
+		if err == nil {
+			break
+		} else {
+			if attempt == maxAttempts {
+				return err
+			}
+		}
+
+		invocationSequence, err := cp.getInvocationSequence(visualizerDBPayload.MigrationUUID,
+			visualizerDBPayload.MigrationPhase)
+
+		if err != nil {
+			log.Warnf("Cannot get invocation sequence for visualization metadata. %s", err)
+			return err
+		}
+
+		timestamp := time.Now().Format("2006-01-02 15:04:05")
+
+		visualizerDBPayload.InvocationSequence = invocationSequence
+		visualizerDBPayload.InvocationTimestamp = timestamp
+	}
+
+	return nil
+}
+
+// Send Table metrics
+func (cp *YugabyteD) sendVisualizerTableMetrics(
+	visualizerTableMetricsList []VisualizerTableMetrics) error {
+
+	cmd := fmt.Sprintf("INSERT INTO %s ("+
+		"migration_uuid, "+
+		"table_name, "+
+		"schema_name,"+
+		"migration_phase, "+
+		"status, "+
+		"count_live_rows, "+
+		"count_total_rows, "+
+		"invocation_timestamp"+
+		") VALUES ", YUGABYTED_TABLE_METRICS_TABLE_NAME)
+
+	var valuesList []string
+	for _, visualizerTableMetrics := range visualizerTableMetricsList {
+		value := fmt.Sprintf("('%s', '%s', '%s', %d, %d, %d, %d, '%s')",
+			visualizerTableMetrics.MigrationUUID,
+			visualizerTableMetrics.TableName,
+			visualizerTableMetrics.Schema,
+			visualizerTableMetrics.MigrationPhase,
+			visualizerTableMetrics.Status,
+			visualizerTableMetrics.CountLiveRows,
+			visualizerTableMetrics.CountTotalRows,
+			visualizerTableMetrics.InvocationTimestamp)
+
+		valuesList = append(valuesList, value)
+	}
+
+	cmd += strings.Join(valuesList, ",")
+
+	cmd += fmt.Sprintf(" ON CONFLICT (migration_uuid, table_name, schema_name, migration_phase) " +
+		"DO UPDATE " +
+		"SET " +
+		"status = EXCLUDED.status," +
+		"count_live_rows = EXCLUDED.count_live_rows," +
+		"count_total_rows = EXCLUDED.count_total_rows," +
+		"invocation_timestamp = EXCLUDED.invocation_timestamp;")
+
+	err := cp.executeCmdOnTarget(cmd)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (cp *YugabyteD) executeInsertQuery(cmd string,
+	visualizerDBPayload VisualizerDBPayload) error {
+
+	cp.Mutex.Lock()
+	defer cp.Mutex.Unlock()
+
+	var err error
+
+	log.Infof("Executing on target DB: [%s] for [%+v]", cmd, visualizerDBPayload)
+	conn, err := cp.getConn()
+	if err != nil {
+		return err
```

Comment:
Please wrap the errors with proper context msg, else it will be very difficult to understand the exact path in which certain error occurred.